src/eth_provider/provider.rs

```diff
@@ -572,6 +584,31 @@ where
         Ok(hash)
     }
 
+    async fn check_cached_balance(&self, signer: Address, transaction_value: U256) -> Result<(), EthApiError> {
```

Comment:
I like the approach a lot, can we just write a simple series of test cases to ensure that this works properly?
- with a balance not in the map and an error
- with a balance in the map and an error
- with a balance not in the map and no error
- with a balance in the map and no error
- ...

Reply:
Tests added. To avoid code repetition, the case with a `balance not in the map and no error` has been added in the `test_send_raw_transaction`, while all the others are in the `test_send_raw_transaction_not_check_cached_balance`